support/ci/shared_build_environment.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+sudo hab license accept
+
+# shellcheck source=./shared.sh
+source ./support/ci/shared.sh
+
+readonly OG_PATH=$PATH
+toolchain=$(get_toolchain)
+eval "$(hab pkg env core/rust/"$toolchain")"
+PATH=$PATH:$OG_PATH
+
+# install_hab_pkg core/rust/"$toolchain"
+sudo -E hab pkg install core/rust/1.79.0/20250606210134 --channel unstable
+install_hab_pkg core/glibc
+install_hab_pkg core/gcc-base
+install_hab_pkg core/binutils
+install_hab_pkg core/cmake
+install_hab_pkg core/bash
+install_hab_pkg core/coreutils
+install_hab_pkg core/curl
+install_hab_pkg core/diffutils
+install_hab_pkg core/gawk
+install_hab_pkg core/git
+install_hab_pkg core/grep
+install_hab_pkg core/libarchive
+install_hab_pkg core/libb2
+install_hab_pkg core/libsodium
+install_hab_pkg core/make
+install_hab_pkg core/openssl
+install_hab_pkg core/pkg-config
+install_hab_pkg core/postgresql17
+install_hab_pkg core/protobuf
+install_hab_pkg core/sed
+install_hab_pkg core/sudo
+install_hab_pkg core/zeromq
+install_hab_pkg core/zlib
```

Comment:
This seems like it has quite a few more packages than we were using in the past (eg libb2, sed, etc). Are we sure we need all of these?